util/rands/cheap.go

```diff
@@ -0,0 +1,67 @@
+// Copyright (c) Tailscale Inc & AUTHORS
+// SPDX-License-Identifier: BSD-3-Clause
+
+package rands
+
+import (
+	exprand "golang.org/x/exp/rand"
+)
+
+// A Rand is a source of random numbers. It is a fork of math/rand and
+// x/exp/rand that is cheap to allocate on the stack that won't escape when used
+// locally, and has a low cost of seed.
+type Rand struct {
+	src exprand.PCGSource
+}
+
+// New returns a new Rand that uses random values from src
+// to generate other random values.
+func New() Rand {
+	var r Rand
+	r.Seed(1)
```

Comment:
Should we select a `crypto/rand`-random seed on creation? Just so the base case is safe? On Linux at least this should be a pretty cheap operation...

If not, a comment calling out that the initial value is fixed and will always return the same sequence would probably be a good idea.

Reply:
I want this to be extremely cheap to create on the stack, and seed in-place, so avoiding a syscall even if it's just to the vdso is desirable - I'll add a comment.

Reply:
As discussed added a mandatory seed to New